### PR TITLE
fix arRPC (again)

### PR DESCRIPTION
**Pull request by ArjixWasTaken**
This time I didn't modify the bridge example much, I only made it use `ArmCordRPC.listen` instead of opening a WebSocket.

https://github.com/OpenAsar/arrpc/blob/main/examples/bridge_mod.js

### CODE REVIEW

**Comment by SMOLKEYS** [comment]
nom

**Comment by ArjixWasTaken** [comment]
Note: The reason behind the `setInterval` is that somehow, the variable `windowCallback` in the preload context either gets garbage collected or is overwritten by `null`.

Setting the callback listener is not a heavy process, so it should be fine to have it re-run every 30 seconds.
A proper fix would be to use Electron's IPC to notify the renderer that `windowCallback` is null

File: src/content/js/rpc.js [hunk 1]
```diff
@@ -1,97 +1,95 @@
...
+                            Dispatcher = candidate;
+                            break;
+                        }
+                    }
+
+                    if (Dispatcher) break;
+                }
+
+                const factories = wpRequire.m;
+                for (const id in factories) {
```
Comment (Vendicated) [inline src/content/js/rpc.js]:
```js
lookupAsset = Object.values(wpRequire.c).find(m => m.exports?.fetchAssetIds).exports.fetchAssetIds;
```

File: src/content/js/rpc.js [hunk 2]
```diff
@@ -1,97 +1,95 @@
...
+                            (e) => typeof e === "function" && e.toString().includes("APPLICATION_ASSETS_FETCH_SUCCESS")
+                        );
+                        if (_lookupAsset)
+                            lookupAsset = async (appId, name) => (await _lookupAsset(appId, [name, undefined]))[0];
+                    }
+
+                    if (lookupAsset) break;
+                }
+
+                for (const id in factories) {
```
Comment (Vendicated) [inline src/content/js/rpc.js]:
```js
lookupApp = Object.values(wpRequire.c).find(m => m.exports?.fetchApplicationsRPC).exports.fetchApplicationsRPC;
```

Comment (ArjixWasTaken) [reply src/content/js/rpc.js]: 🙏

Comment (nyabinary) [reply src/content/js/rpc.js]: wha does t hat mean?

Comment (ArjixWasTaken) [reply src/content/js/rpc.js]: The pray emoji?

It means thx

**Review by Hellscaped — APPROVED** [review]
looks fine to me

**Comment by SMOLKEYS** [comment]
image

:husk:

**Review by SMOLKEYS — APPROVED** [review]

**Pull request merged**